en/lessons/specifics/ecto/03_associations.md

```diff
@@ -0,0 +1,567 @@
+
+```bash
+mix ecto.gen.migration actor
+```
+
+Define the migration:
+
+```Elixir
+# priv/migrations/*_create_actor.ex
+
+defmodule Moviedb.Repo.Migrations.Actor do
+  use Ecto.Migration
+
+  def change do
+    create table(:actors) do
+      add :name, :string
+    end
+  end
+end
+```
+
+Let's generate our join table migration:
+
+```bash
+mix ecto.gen.migration create_movies_actors
+```
+
+We'll define our migration such that the table has two foreign keys. We'll also add a unique index to enforce unique pairings of actors and movies:
+
+```elixir
+# priv/migrations/*_create_movie_actors.ex
+
+defmodule Moviedb.Repo.Migrations.CreateMovieActors do
+  use Ecto.Migration
+
+  def change do
+    create table(:movies_actors) do
+      add :movie_id, references(:movies)
+      add :actor_id, references(:actors)
+    end
+
+    create unique_index(:movies_actors, [:movie_id, :actor_id])
+  end
+end
+```
+
+Next up, let's add the `many_to_many` macro to our `Movies` schema:
+
+```elixir
+# lib/movie.ex
+
+defmodule Moviedb.Movie do
+  use Ecto.Schema
+
+  schema "movies" do
+    field :title, :string
+    field :tagline, :string
+    has_many :characters, Moviedb.Character
+    has_one :distributor, Moviedb.Distributor
+    many_to_many :actors, Moviedb.Actor, join_through: "movies_actors" # I'm new!
+  end
+end
+```
+
+Finally, we'll define our `Actor` schema with the same `many_to_many` macro.
+
+```elixir
+# lib/actor.ex
+
+defmodule Moviedb.Movie do
+  use Ecto.Schema
+
+  schema "actors" do
+    field :name, :string
+    many_to_many :movies, Moviedb.Movie, join_through: "movies_actors"
+  end
+end
+```
+
+We're ready to run our migrations:
+
+```bash
+mix ecto.migrate
+```
+
+## Saving Associated Data
+
+The manner in which we save records along with their associated data depends on the nature of the relationship between the records. Let's start with the "Belongs to/has many" relationship.
+
+### Belongs To
+#### Saving With `Ecto.build_assoc/3`
+With a `belongs_to` relationship, we can leverage Ecto's `build_assoc` function.
+
+[`build_assoc`](https://hexdocs.pm/ecto/Ecto.html#build_assoc/3) takes in three arguments:
+
+* The struct of the record we want to save.
+* The name of the association.
+* Any attributes we want to assign to the associated record we are saving.
+
+Let's save a movie and and associated character:
+
+First, we'll create a movie record:
+
+```elixir
+iex> alias Moviedb.{Movie, Character, Repo}
+iex> movie = %Movie{title: "Ready Player One", tagline: "Something about video games"}
+
+%Moviedb.Movie{
+  __meta__: #Ecto.Schema.Metadata<:built, "movies">,
+  actors: #Ecto.Association.NotLoaded<association :actors is not loaded>,
+  characters: #Ecto.Association.NotLoaded<association :characters is not loaded>,
+  distributor: #Ecto.Association.NotLoaded<association :distributor is not loaded>,
+  id: nil,
+  tagline: "Something about video games",
+  title: "Ready Player One"
+}
+
+iex> movie = Repo.insert!(movie)
```

Comment:
I'm going to skip this for now because I'll need to rebuild the repo with this code just to run that one line to get the output. Unless you want to fake it in which case be my guest :)